jigglypuffRL/common/values.py

```diff
@@ -34,7 +34,39 @@ def __init__(self, state_dim, action_dim=None, val_type="V", hidden=(32, 32)):
         self.model = _get_val_model(mlp, val_type, state_dim, hidden, action_dim)
 
 
-value_registry = {"mlp": MlpValue}
+def CNNValue(BaseValue):
+    """
+    CNN Value
+    :param state_dim: (int) state dimension of environment
+    :param action_dim: (int) action dimension of environment
+    :param history_length: (int) length of history of states
+    :param val_type: (str) type of value function.
+        'V' for V(s), 'Qs' for Q(s), 'Qsa' for Q(s,a)
+    :param hidden: (tuple or list) sizes of hidden layers
+    :param disc: (bool) discrete action space?
+    """
+    def __init__(
+        self, state_dim, action_dim, history_length=4, val_type="Qs",
+        hidden=(256, 256), disc=True
+    ):
+        super(CNNValue, self).__init__(disc, state_dim, action_dim, hidden)
```

Comment:
Your error is arising from here. This will the Parent class' constructor (`BaseValue`) and it does will not take any arguments

Reply:
The mistake was I used `def CNNValue(BaseValue):` in place of `class CNNValue(BaseValue):` 😅

Reply:
Lol